### PR TITLE
Develop

### LINKED ISSUE #4: Explore the Entity AttributeExact Match Configuration
The [Entity AttributeExact Match Configuration](https://shibboleth.atlassian.net/wiki/spaces/IDP5/pages/3199502013/EntityAttributeExactMatchConfiguration) is a PolicyRule that returns true if the SAML metadata for the requester contains [<mdattr:EntityAttribute>](http://docs.oasis-open.org/security/saml/Post2.0/sstc-metadata-attr.html) extension data matching the supplied parameterization.

Reference: [SAML V2.0 Subject Identifier Attributes Profile Version 1.0](http://docs.oasis-open.org/security/saml-subject-id-attr/v1.0/csprd03/saml-subject-id-attr-v1.0-csprd03.html)

### LINKED ISSUE #5: Default attribute filter
The default attribute filter has a number of active policies that release values that may not meet the AAF attribute release methods. For example a rule exists to release the "schacHomeOrganization" to everybody.

Review these rules to determine if they are appropriate for AAF IdPs.

### LINKED ISSUE #6: Upgrade the attribute resolver
Review and upgrade the attribute resolver based on v5 best practice. 

Ensure all scripts are based on Nashorn.

New values for the eduPersonAffiliation.

New attributes: subjectID and pairwiseID.

**Core Attributes**
- [ ] auEduPersonSharedToken
- [x] displayName
- [x] eduPersonAffiliation
- [x] eduPersonAssurance
- [x] eduPersonEntitlement
- [x] eduPersonPrincipalName
- [x] eduPersonScopedAffiliation
- [x] eduPersonTargetedID
- [x] mail
- [x] givenName
- [x] schacHomeOrganization
- [x] schacHomeOrganizationType
- [x] o
- [x] sn
- [x] samlSubjectID
- [x] samlPairwiseID

**Conditional Attributes**
- [x] eduPersonOrcid

**Optional Attributes**
- [ ] auEduPersonLegalName
- [ ] businessCategory
- [ ] carLicense
- [x] cn
- [ ] departmentNumber
- [ ] description
- [ ] division
- [x] dsdStudentID
- [ ] eduPersonPrimaryAffiliation
- [ ] employeeNumber
- [ ] employeeType
- [ ] facsimileTelephoneNumber
- [ ] initials
- [ ] l
- [ ] manager
- [ ] mobileNumber
- [ ] organizationalUnit
- [ ] physicalDeliveryOfficeName
- [ ] postOfficeBox
- [ ] postalAddress
- [ ] postalCode
- [ ] preferredLanguage
- [ ] schacPersonalUniqueID
- [ ] seeAlso
- [ ] st
- [ ] street
- [ ] telephoneNumber
- [ ] title
- [x] uid

Data Connectors
- [x] staticAttributes
- [x] ldap
- [x] StoredIDConnector
- [ ] aafSharedToken
- [x] computed

### LINKED ISSUE #8: New eduGAIN attribute filter
Update the eduGAIN attribute filter to support new entity categories

- Research and Scholarshop 1.3
- Anonymous Access v.1 and 2
- Pseudonymous Access v.1 and 2
- Personalized Access V.1 and 2
- Code of Conduct v2

### LINKED ISSUE #9: Refresh the federation attribute filter
Ensure the federation attribute files is up to date.

- New attributes and schema

### LINKED ISSUE #10: Reduce the frequency of Metadata retrival
Currently the refreshing federation and eduGAIN metadata is too frequent. Update the following parameters to requce the frequency. 

`maxRefreshDelay` to `PT4H` for AAF amd 'PT24H' for eduGAIN. Upper bound on the next refresh from the time calculated based on the metadata's expiration.
`refreshDelayFactor `to `0.75`. A factor applied to the initially determined refresh time in order to determine the next refresh time (typically to ensure refresh takes place prior to the metadata's expiration). Attempts to refresh metadata will generally begin around the product of this number and the maximum refresh delay.

Current settings of 'PT2H' and '0.125' have IdP reloading around every 20 minutes.

### LINKED ISSUE #12: Remove the export-v3-config.sh script
No longer require, so will remove to aviod confusing users.

### LINKED ISSUE #13: Upgrade to V5.1.3
Currently unreleased, but coming soon. https://shibboleth.atlassian.net/wiki/spaces/IDP5/pages/3199500367/ReleaseNotes#5.1.0-(Unreleased)

### LINKED ISSUE #14: -y option for deploy.sh
All the deploy to run uninterrupted, no questions if -y option is set.

### PR DESCRIPTION
**Pull request by trsau**
Ready for production use.

Resolves #15 
Resolves #13 
Resolves #14 
Resolves #12 
Resolves #10 
Resolves #9 
Resolves #8 
Resolves #6 
Resolves #5 
Resolves #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-For a full set of documentation and guidance on upgrading from Shibboleth version 3 please refer to the [AAF IdPv4 Installer Knowledge base](https://aaf.freshdesk.com/support/solutions/articles/19000120020-shibboleth-idpv4-installer).
+For a full set of documentation and guidance on upgrading from Shibboleth version 4 please refer to the [AAF IdPv4 Installer Knowledge base](https://support.aaf.edu.au/support/solutions/articles/19000159910-shibboleth-idp-version-5-installer).
 
 ## Status
 This software is actively being developed and maintained. It is ready for use in production environments.
 
 This version now enables the technical connection to eduGAIN, the global federation of federations.
 
-This release is for Shibboleth version 4.3.1 running on Jetty 9.4.51
+This release is for Shibboleth version 5.1.3 running on Jetty 12.0.12
```
Comment (ArthurZheng) [inline README.md:18]: Just FYI: Jetty has a newer release, 12.0.13 coming soon :-)
There's a bug (Scott raised it - JSP temp directory regression, possibly due to fix for #12044 #12124) in 12.0.12

File: Whats-new.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# What's new in the AAF IdP Installer
+
+## Overview
+This is a brief not of the new features and functionality provided in the AAF Installer.
```
Comment (ArthurZheng) [inline Whats-new.md]:
```suggestion
This is a brief note of the new features and functionality provided in the AAF Installer.
```

File: assets/idp.example.edu.dist/idp/conf/relying-party.xml [hunk 3]
```diff
@@ -47,7 +50,6 @@
                 <ref bean="SAML2.AttributeQuery" />
                 -->
                 <ref bean="SAML2.ArtifactResolution" />
-                <ref bean="Liberty.SSOS" />
```
Comment (ArthurZheng) [inline assets/idp.example.edu.dist/idp/conf/relying-party.xml:50]: yeah, this is removed from Shib 5

File: assets/idp.example.edu.dist/idp/conf/services.xml [hunk 4]
```diff
@@ -37,9 +37,7 @@
     <util:list id ="shibboleth.AttributeFilterResources">
         <value>%{idp.home}/conf/metadata-based-attribute-filter.xml</value>
         <value>%{idp.home}/conf/attribute-filter.xml</value>
-        {% if enable_edugain == "true" %}
```
Comment (ArthurZheng) [inline assets/idp.example.edu.dist/idp/conf/services.xml:40]: I was just wondering why the removal of edugain here?

File: site_v5.yml [hunk 5]
```diff
@@ -43,67 +43,77 @@
...
-        sha256sum: 8462a7d983a10700c2cc559b7df4254f3a21c2611c32476cb5a6702ce51524c9
+      jcl_over_slf4j:
+        baseurl: "{{ aaf_binaries.baseurl }}/jars/slf4j"
+        version: 2.0.5
+        sha256sum: 95fa90eb540df07b6a2fa70f3613761180d07c9c8f1c299e3f9a5c943313a2d6
       logback_classic:
         baseurl: "{{ aaf_binaries.baseurl }}/jars/logback"
-        version: 1.2.10
-        sha256sum: 3160ae988af82c8bf3024ddbe034a82da98bb186fd09e76c50543c5b9da5cc5e
+        version: 1.3.6
```
Comment (ArthurZheng) [inline site_v5.yml:59]: It seems that Jetty 11/12 uses logback-classic 1.5.6

https://mvnrepository.com/artifact/ch.qos.logback/logback-classic/1.5.6 (Released on Apr 17, 2024)
https://repo1.maven.org/maven2/ch/qos/logback/logback-classic/1.5.6/logback-classic-1.5.6.jar

File: site_v5.yml [hunk 6]
```diff
@@ -43,67 +43,77 @@
...
-        sha256sum: ba51a3fe56691f9dd7fe742e4a73c3ab4aaaa32202c73409ba56f18687399a08
-      jcl_over_slf4j:
-        baseurl: "{{ aaf_binaries.baseurl }}/jars/slf4j"
-        version: 1.7.33
-        sha256sum: a2c0b70507e6f769ea571608fd9b728329c5c723af37c3b772127c58fb593dfc
-      slf4j_api:
-        baseurl: "{{ aaf_binaries.baseurl }}/jars/slf4j"
-        version: 1.7.33
-        sha256sum: 336811855ee97fcd8413343dc42f4d26362502860e8a9193e4db305e3e8ef80a
+        version: 1.3.6
```
Comment (ArthurZheng) [inline site_v5.yml:63]: It seems that Jetty 11/12 uses logback-core 1.5.6
https://mvnrepository.com/artifact/ch.qos.logback/logback-core/1.5.6 (Released on Apr 17, 2024)
https://repo1.maven.org/maven2/ch/qos/logback/logback-core/1.5.6/logback-core-1.5.6.jar

File: site_v5.yml [hunk 7]
```diff
@@ -43,67 +43,77 @@
...
+        url: "{{ download.jetty.baseurl }}/jetty-home-{{ download.jetty.version }}.tar.gz"
       shib_idp:
         url: "{{ download.shib_idp.baseurl }}/shibboleth-identity-provider-{{ download.shib_idp.version }}.tar.gz"
       mysql_connector:
         url: "{{ download.mysql_connector.baseurl }}/mysql-connector-java-{{ download.mysql_connector.version }}.tar.gz"
       dta_ssl:
         url: "{{ download.dta_ssl.baseurl }}/jetty94-dta-ssl-{{download.dta_ssl.version}}.jar"
       aaf_shib_ext:
-        url: "{{ download.aaf_shib_ext.baseurl }}/aaf-shib-ext-{{ download.aaf_shib_ext.version }}.jar"
+        url: "{{ download.aaf_shib_ext.baseurl }}/aaf-shibext-{{ download.aaf_shib_ext.version }}.jar"
```
Comment (ArthurZheng) [inline site_v5.yml:89]: Is this a typo?
```suggestion
        url: "{{ download.aaf_shib_ext.baseurl }}/aaf-shi-bext-{{ download.aaf_shib_ext.version }}.jar"
```

File: tasks/libs.yml [hunk 8]
```diff
@@ -78,40 +63,28 @@
...
     state: link
 
-- name: 'Install slf4j-api'
+- name: 'Install aaf-shibext library'
   get_url:
-    url: '{{ urls.slf4j_api.url }}'
-    dest: '{{ shib_idp.libs }}/slf4j-api-{{ download.slf4j_api.version }}.jar'
-    checksum: 'sha256:{{ download.slf4j_api.sha256sum }}'
+    url: '{{ urls.aaf_shib_ext.url }}'
+    dest: '{{ shib_idp.home }}/edit-webapp/WEB-INF/lib/aaf-shibext-{{ download.aaf_shib_ext.version }}.jar'
```
Comment (ArthurZheng) [inline tasks/libs.yml:73]: I guess, not a typo, given it appears twice
```suggestion
    dest: '{{ shib_idp.home }}/edit-webapp/WEB-INF/lib/aaf-shib-ext-{{ download.aaf_shib_ext.version }}.jar'
```

**Review by ArthurZheng — APPROVED** [review]
Great work @trsau 👍

File: README.md (hunk 1 above)
Comment (trsau) [reply README.md:18]: Will update when it becomes available.

File: assets/idp.example.edu.dist/idp/conf/services.xml (hunk 4 above)
Comment (trsau) [reply assets/idp.example.edu.dist/idp/conf/services.xml:40]: The eduGAIN attribute filter just covered the R&S entity category. This have been replaced with a policy-based-attribute filter the handles all of the REFEDS entity categories.

File: site_v5.yml (hunk 5 above)
Comment (trsau) [reply site_v5.yml:59]: Have added a new issue to upgrade to latest version.

File: site_v5.yml (hunk 6 above)
Comment (trsau) [reply site_v5.yml:63]: Same issue as logback-classic

File: site_v5.yml (hunk 7 above)
Comment (trsau) [reply site_v5.yml:89]: Not a typo. New version based to the NZ code, so new directory.

File: tasks/libs.yml (hunk 8 above)
Comment (trsau) [reply tasks/libs.yml:73]: Not a typo.

**Pull request merged**